pkgs/tools/package-management/guix/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv, pkgs, lib, fetchurl, pkg-config, makeWrapper, zlib, bzip2, guile
+, guile-gcrypt, guile-git, guile-json, guile-sqlite3, guile-ssh, gnutls
+, storeDir ? null, stateDir ? null }:
+
+let
+  guile-gnutls = (gnutls.override {
+    inherit guile;
+    guileBindings = true;
+  }).overrideAttrs (attrs: {
+    # Workaround until https://github.com/NixOS/nixpkgs/pull/80206 is allowed in
+    configureFlags = [
+      "--with-guile-site-dir=\${out}/share/guile/site"
+      "--with-guile-site-ccache-dir=\${out}/share/guile/ccache"
+      "--with-guile-extension-dir=\${out}/lib/guile/extensions"
+    ];
+  });
+in stdenv.mkDerivation rec {
+  pname = "guix";
+  version = "1.1.0";
+
+  src = fetchurl {
+    url = "mirror://gnu/guix/${pname}-${version}.tar.gz";
+    sha256 = "03v48cq94678pi7bkdjfwksrq40lr6q38s38jxh7qjdcyipi0naw";
+  };
+
+  postConfigure = ''
+    sed -i '/guilemoduledir\s*=/s%=.*%=''${out}/share/guile/site%' Makefile;
+    sed -i '/guileobjectdir\s*=/s%=.*%=''${out}/share/guile/ccache%' Makefile;
+  '';
+
+  modules = [
+    guile-gcrypt guile-git guile-json guile-sqlite3 guile-ssh guile-gnutls.out guile-git.bytestructures
+  ];
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
+  buildInputs = [ zlib bzip2 ] ++ modules;
+  propagatedBuildInputs = [ guile ];
+
+  GUILE_LOAD_PATH = let
+    guilePath = [
+      "\${out}/share/guile/site"
```

Comment:
Won't this try to reference `$out` at runtime?

Reply:
No, it falls through to be evaluated while creating the wrappers in installPhase. There may be a better way of quoting them, though, I'm not sure

Reply:
Maybe `placeholder "out"` would be better. At least that way its more obvious.